connection/types.go

```diff
@@ -28,32 +27,34 @@ const (
 
 // probably needs a better name since its not only used for piping the connection
 type PipeConnection interface {
+	getConn() ByteConnection
+}
+
+type ByteConnection interface {
 	io.Writer
 	io.Reader
+	io.Closer
 }
 
 type Connection interface {
 	infrared.PacketWriter
 	infrared.PacketReader
-	PipeConnection
 }
```

Comment:
Maybe we should just move the PacketWriter and PacketReader over from the old Conn and delete it